tests/utilities.cpp

```diff
@@ -1051,16 +1069,48 @@ void toQureg(Qureg qureg, QMatrix mat) {
     copyStateToGPU(qureg);
 }
 
-void setRandomPauliSum(qreal* coeffs, pauliOpType* codes, int numQubits, int numTerms) {
```

Comment:
`setRandomPauliSum` and `setRandomDiagonalPauliSum` should be two separate functions.

Generally, if your function contains a giant if ..:
```C++
f(x) {
    if (x == A) { 
        // A stuff 
    } else {
        // B stuff 
    }
}
```
then it should really become
```C++
fA() {
    // A stuff 
}
fB() {
    // B stuff
}
```